src/languages/typescript.ts

```diff
@@ -135,55 +147,68 @@ const nodeMatchers: Record<ScopeType, NodeMatcher> = {
-  class: possiblyExportedDeclaration("class_declaration", "class"),
-  statement: possiblyExportedDeclaration(...STATEMENT_TYPES),
-  arrowFunction: hasType("arrow_function"),
-  functionCall: hasType("call_expression", "new_expression"),
+  ifStatement: typeMatcher("if_statement"),
+  class: matcher(possiblyExportedDeclaration("class_declaration", "class")),
+  statement: matcher(possiblyExportedDeclaration(...STATEMENT_TYPES)),
+  arrowFunction: typeMatcher("arrow_function"),
+  functionCall: typeMatcher("call_expression", "new_expression"),
+  functionName: cascadingMatcher(
+    composedMatcher([
+      findNodeOfType("function_declaration", "method_definition"),
+      getNameNode,
+    ]),
+    composedMatcher([findClassPropertyArrowFunction, getNameNode]),
+    composedMatcher([findNamedArrowFunction, getNameNode])
+  ),
   type: cascadingMatcher(
     // Typed parameters, properties, and functions
-    childNodeMatcher(getTypeNode, getNodeWithLeadingDelimiter),
-
+    matcher(findTypeNode, selectWithLeadingDelimiter),
     // Type alias/interface declarations
-    possiblyExportedDeclaration(
-      "type_alias_declaration",
-      "interface_declaration"
+    matcher(
+      possiblyExportedDeclaration(
+        "type_alias_declaration",
+        "interface_declaration"
+      )
     )
   ),
-  argumentOrParameter: delimitedMatcher(
-    (node) =>
-      (node.parent?.type === "arguments" &&
-        (isExpression(node) || node.type === "spread_element")) ||
-      node.type === "optional_parameter" ||
-      node.type === "required_parameter",
-    (node) => node.type === "," || node.type === "(" || node.type === ")",
-    ", "
+  argumentOrParameter: matcher(
+    findNode(
+      (node) =>
+        (node.parent?.type === "arguments" &&
+          (isExpression(node) || node.type === "spread_element")) ||
+        node.type === "optional_parameter" ||
+        node.type === "required_parameter"
+    ),
+    selectDelimited(
+      (node) => node.type === "," || node.type === "(" || node.type === ")",
+      ", "
+    )
   ),
   namedFunction: cascadingMatcher(
     // Simple case, eg
     // function foo() {}
-    possiblyExportedDeclaration("function_declaration", "method_definition"),
+    matcher(
+      possiblyExportedDeclaration("function_declaration", "method_definition")
+    ),
 
     // Class property defined as field definition with arrow
     // eg:
     // class Foo {
     //   bar = () => "hello";
     // }
-    (editor: TextEditor, node: SyntaxNode) =>
-      node.type === "public_field_definition" &&
-      getValueNode(node)!.type === "arrow_function"
-        ? simpleSelectionExtractor(node)
-        : null,
+    matcher(findClassPropertyArrowFunction),
 
     // eg:
     // const foo = () => "hello"
-    possiblyWrappedNode(
-      (node) => node.type === "export_statement",
-      isNamedArrowFunction,
-      (node) => [getDeclarationNode(node)]
+    matcher(
+      findPossiblyWrappedNode(
```

Comment:
I wonder why this one can't use `possiblyExportedDeclaration` 🤔.  Looks like it was already this way but is a bit curious